llama_index/llms/huggingface.py

```diff
@@ -294,3 +311,243 @@ def stream_chat(
         prompt = self._messages_to_prompt(messages)
         completion_response = self.stream_complete(prompt, formatted=True, **kwargs)
         return stream_completion_response_to_chat_response(completion_response)
+
+
+def chat_messages_to_conversational_kwargs(
+    messages: Sequence[ChatMessage],
+) -> Dict[str, Any]:
+    """Convert ChatMessages to keyword arguments for Inference API conversational."""
+    if len(messages) % 2 != 1:
+        raise NotImplementedError("Messages passed in must be of odd length.")
+    last_message = messages[-1]
+    kwargs: Dict[str, Any] = {
+        "text": last_message.content,
+        **last_message.additional_kwargs,
+    }
+    if len(messages) != 1:
+        kwargs["past_user_inputs"] = []
+        kwargs["generated_responses"] = []
+        for user_msg, assistant_msg in zip(messages[::2], messages[1::2]):
+            if (
+                user_msg.role != MessageRole.USER
+                or assistant_msg.role != MessageRole.ASSISTANT
+            ):
+                raise NotImplementedError(
+                    "Didn't handle when messages aren't ordered in alternating"
+                    f" pairs of {(MessageRole.USER, MessageRole.ASSISTANT)}."
+                )
+            kwargs["past_user_inputs"].append(user_msg.content)
+            kwargs["generated_responses"].append(assistant_msg.content)
+    return kwargs
+
+
+def conversational_output_to_chat_response(
+    output: "ConversationalOutput", role: MessageRole = MessageRole.ASSISTANT
+) -> ChatResponse:
+    return ChatResponse(
+        message=ChatMessage(role=role, content=output["generated_text"])
+    )
+
+
+class Pooling(str, Enum):
```

Comment:
Does it make more sense to have this in an embeddings util file? Would be useful for the existing huggingface and optimum embeddings classes

Reply:
Yeah good suggestion, will move it there

Reply:
Done